mongofiles/mongofiles_test.go

```diff
@@ -476,67 +484,124 @@ func TestMongoFilesCommands(t *testing.T) {
+
+			testFiles := make([]string, numCopies)
+			for i := 0; i < numCopies; i++ {
+				newFile := util.ToUniversalPath("testdata/" + fmt.Sprintf(copyFormat, i))
+
+				// Makes new copies of lorem ipsum file
+				err = func(src, dst string) error {
+					in, err := os.Open(src)
+					if err != nil {
+						return err
+					}
+					defer in.Close()
+
+					out, err := os.Create(dst)
+					if err != nil {
+						return err
+					}
+					defer out.Close()
+
+					_, err = io.Copy(out, in)
+					if err != nil {
+						return err
+					}
+
+					return nil
+				}(originalFile, newFile)
+				So(err, ShouldBeNil)
+
+				testFiles[i] = newFile
+			}
+
+			mf, err := simpleMongoFilesInstanceWithMultipleFileNames("put", testFiles...)
 			So(err, ShouldBeNil)
-			So(mf, ShouldNotBeNil)
-			mf.StorageOptions.LocalFileName = util.ToUniversalPath("testdata/" + filename)
 
 			var buff bytes.Buffer
 			log.SetWriter(&buff)
 
-			Convey("insert the file by creating two chunks (ceil(287,613 / 255 * 1024)) in GridFS", func() {
-				buff.Truncate(0)
-				str, err := mf.Run(false)
+			str, err := mf.Run(false)
+			So(err, ShouldBeNil)
+			So(str, ShouldBeEmpty)
+
+			Convey("log an event specifying the completion of each file", func() {
+				const (
+					logAdding = "adding gridFile: %v"
+					logAdded  = "added gridFile: %v"
+				)
+
+				logOutput := buff.String()
+
+				for _, testFile := range testFiles {
+					So(logOutput, ShouldContainSubstring, fmt.Sprintf(logAdding, testFile))
+					So(logOutput, ShouldContainSubstring, fmt.Sprintf(logAdded, testFile))
+
+				}
+			})
+
+			Convey("and files should exist in GridFS", func() {
+				bytesGotten, err := getFilesAndBytesListFromGridFS()
 				So(err, ShouldBeNil)
-				So(str, ShouldEqual, "")
-				So(buff.Len(), ShouldNotEqual, 0)
 
-				Convey("and files should exist in gridfs", func() {
-					bytesGotten, err := getFilesAndBytesListFromGridFS()
-					So(err, ShouldBeNil)
-					So(len(bytesGotten), ShouldEqual, len(testFiles)+1)
-					So(bytesGotten[filename], ShouldEqual, 287613)
-					So(bytesGotten, ShouldContainKey, filename)
-				})
+				for _, testFile := range testFiles {
+					So(bytesGotten, ShouldContainKey, testFile)
+					So(bytesGotten[testFile], ShouldEqual, numBytes)
+				}
+			})
+
+			Convey("and each file should have exactly the same content as the original file", func() {
+				loremIpsumOrig, err := os.Open(util.ToUniversalPath("testdata/lorem_ipsum_287613_bytes.txt"))
+				So(err, ShouldBeNil)
+				defer loremIpsumOrig.Close()
 
-				Convey("and should have exactly the same content as the original file", func() {
-					buff.Truncate(0)
-					mfAfter, err := simpleMongoFilesInstanceWithFilename("get", "lorem_ipsum_287613_bytes.txt")
+				const localFileName = "lorem_ipsum_copy.txt"
+				buff.Truncate(0)
+				for _, testFile := range testFiles {
+					mfAfter, err := simpleMongoFilesInstanceWithFilename("get", testFile)
```

Comment:
I thought in this test, you gonna test multiple files in one `Run`?

Reply:
This is the test case for the polyadic `--put` subcommand. Here, `mfAfter` is meant to `--get` the files after applying `--put` so that we can verify that they were uploaded correctly. I `--get` them one by one, so that I avoid unfortunate naming collisions within the files.

This might not be necessary if I use different _lorem ipsum_ files, as I suggested in your other comment -- in that case, I would be able to use the polyadic `--get` for testing.

Reply:
I see what you mean. If this is the put test, then either way works!

Reply:
I do do that -- this test case is intended to test multiple files in one call to `--put`. Here, `mfAfter` is intended only to get the files one-by-one with `--local` to verify that each test file was uploaded properly.

I think I can also use the new polyadic `--get` call to get all the test files at once before looping through each of them, but I don't think it'd make any difference semantically.